astropy/io/fits/file.py

```diff
@@ -83,6 +83,7 @@ class _File(object):
     @deprecated_renamed_argument('clobber', 'overwrite', '2.0')
     def __init__(self, fileobj=None, mode=None, memmap=None, overwrite=False,
                  cache=True):
+        from http.client import HTTPResponse
```

Comment:
It's not critical so I think we can skip Python 2 support, but the point is that merging this in the current state may be problematic when backporting other PRs. So maybe maybe we just wait a bit before merging ?